tools/tesseratos/src/tesseratos/scene_editor/plugin.cpp

```diff
@@ -219,14 +233,28 @@ static void showSceneEntities(std::vector<std::pair<std::string, Entity>>& entit
     ImGui::PopStyleColor(1);
 }
 
+static void updateNameComponent(SceneInfo& scene, Commands& commands)
+{
+    for (const auto& [name, entity] : scene.entities)
+    {
+        auto newName = scene.name + "." + name;
+        commands.add(entity, Name{.value = newName});
+    }
+    for (auto& subscene : scene.scenes)
+    {
+        updateNameComponent(*subscene, commands);
+    }
+}
+
 // Recursively draws the scene hierarchy and allows the user to remove scenes
-static void showSceneHierarchy(SceneInfo& scene, Commands& cmds, EntitySelector& selector, int hierarchyDepth)
+static void showSceneHierarchy(SceneInfo& scene, Commands& cmds, EntitySelector& selector, int hierarchyDepth,
+                               Assets& assets, State& state)
 {
     ImGui::PushID(&scene);
 
     bool nodeOpen;
     bool isTree = false;
-
+    auto* bufferasset = new Asset<Scene>;
```

Comment:
Don't allocate it like this - this is probably causing memory leaks. Would using a normal variable cause problems here? I.e. `Asset<Scene> asset;`